alert-common/src/main/java/com/synopsys/integration/alert/common/email/EmailMessagingService.java

```diff
@@ -168,22 +168,25 @@ private Session createMailSession(EmailProperties emailProperties) {
     public void sendMessages(EmailProperties emailProperties, Session session, List<Message> messages) throws AlertException {
         Set<String> errorMessages = new HashSet<>();
         Set<String> invalidRecipients = new HashSet<>();
-        for (Message message : messages) {
-            Address[] recipients = null;
-            try {
-                recipients = message.getAllRecipients();
-                if (Boolean.valueOf(emailProperties.getJavamailOption(EmailPropertyKeys.JAVAMAIL_AUTH_KEY))) {
-                    sendAuthenticated(emailProperties, message, session);
-                } else {
-                    Transport.send(message);
-                }
-            } catch (MessagingException e) {
-                if (recipients != null) {
-                    Stream.of(recipients).map(Address::toString).forEach(invalidRecipients::add);
+
+        try (Transport transport = getTransport(emailProperties, session)) {
+            for (Message message : messages) {
+                Address[] recipients = null;
+                try {
```

Comment:
Not a fan of this nested try. Could we instead extract a second public sendMessages method which takes a Transport and a List\<Message\>?